actors/migration/top.go

```diff
@@ -119,54 +158,110 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, stateRootIn cid
 		return cid.Undef, err
 	}
 
-	// Extra actor setup
-
-	// miner
+	// Accumulator setup
 	transferFromBurnt := big.Zero()
 	powerUpdates := &PowerUpdates{
 		claims: make(map[address.Address]power0.Claim),
 		crons:  make(map[abi.ChainEpoch][]power0.CronEvent),
 	}
 
-	// Iterate all actors in old state root
-	// Set new state root actors as we go
-	err = actorsIn.ForEach(func(addr address.Address, actorIn *states.Actor) error {
-		transfer, err := migrateOneActor(ctx, store, addr, *actorIn, actorsOut, priorEpoch, powerUpdates)
-		if err != nil {
-			return err
+	// Setup synchronization
+	grp, ctx := errgroup.WithContext(ctx)
+	inputCh := make(chan *migrationInput)
+	resultCh := make(chan *migrationResult)
+	var workerWg sync.WaitGroup
```

Comment:
uber nit: I'd move this variable down right above the workers, to make it clear it's only used there.